src/vulkan/descriptor.cc

```diff
@@ -79,26 +106,40 @@ void Descriptor::UpdateDescriptorSet(VkDescriptorSet descriptor_set,
     case VK_DESCRIPTOR_TYPE_STORAGE_IMAGE:
     case VK_DESCRIPTOR_TYPE_INPUT_ATTACHMENT:
       write.pImageInfo = image_info;
-      assert(buffer_info == nullptr && texel_view == nullptr);
       break;
+    default:
+      return Result(
+          "Vulkan::UpdateDescriptorSetForImage not descriptor based on image");
+  }
+
+  vkUpdateDescriptorSets(device_, 1, &write, 0, nullptr);
+  return {};
+}
+
+Result Descriptor::UpdateDescriptorSetForBufferView(
+    VkDescriptorSet descriptor_set,
+    VkDescriptorType descriptor_type,
+    const VkBufferView* texel_view) {
+  VkWriteDescriptorSet write = {};
+  write.sType = VK_STRUCTURE_TYPE_WRITE_DESCRIPTOR_SET;
+  write.dstSet = descriptor_set;
+  write.dstBinding = binding_;
+  write.dstArrayElement = 0;
+  write.descriptorCount = 1;
+  write.descriptorType = descriptor_type;
```

Comment:
This setup can be moved to a common method so we don't duplicate it.

Reply:
Done